src/components/dataset/DatasetReader.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Map } from 'immutable';
+import ReactJson from 'react-json-view';
+import { withTranslation } from 'react-i18next';
+import Loader from '../common/Loader';
+import { MAX_FILE_SIZE } from '../../config/constants';
+
+class DatasetReader extends Component {
+  static propTypes = {
+    content: PropTypes.string,
+    size: PropTypes.number,
+    t: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    content: Map(),
+    size: 0,
+  };
+
+  state = {
+    json: null,
+  };
+
+  async componentDidMount() {
+    const { content, size } = this.props;
+    if (size < MAX_FILE_SIZE && content) {
+      this.loadFile();
+    }
+  }
+
+  async componentDidUpdate({ content: prevContent }) {
+    const { content, size } = this.props;
+    if (size < MAX_FILE_SIZE && content !== prevContent) {
+      this.loadFile();
+    }
+  }
+
+  loadFile = () => {
+    const { content } = this.props;
+    const parsedContent = JSON.parse(content);
+    this.setState({ json: parsedContent });
+  };
+
+  render() {
+    const { content, size, t } = this.props;
+    const { json } = this.state;
+
+    if (!content) {
+      return <Loader />;
+    }
+
+    if (size > MAX_FILE_SIZE) {
+      return <span>{t('This file is too huge to be displayed.')}</span>;
```

Comment:
(1) Let's use a Typography element instead of a span.
(2) 'This file is too **big** to be displayed.'